bindings/flow/fdb_flow.actor.cpp

```diff
@@ -98,6 +98,81 @@ void fdb_flow_test() {
 }
 
 namespace FDB {
+	class DatabaseImpl : public Database, NonCopyable {
+	public:
+		virtual ~DatabaseImpl() { fdb_database_destroy(db); }
+
+		Reference<Transaction> createTransaction() override;
+		void setDatabaseOption(FDBDatabaseOption option, Optional<StringRef> value = Optional<StringRef>()) override;
+
+	private:
+		FDBDatabase* db;
+		explicit DatabaseImpl(FDBDatabase* db) : db(db) {}
+
+		friend class API;
+	};
+
+	class TransactionImpl : public Transaction, private NonCopyable, public FastAllocated<TransactionImpl> {
+		friend class DatabaseImpl;
+
+	public:
+		virtual ~TransactionImpl() {
+			if (tr) {
+				fdb_transaction_destroy(tr);
+			}
+		}
+
+		void setReadVersion(Version v) override;
+		Future<Version> getReadVersion() override;
+
+		Future<Optional<FDBStandalone<ValueRef>>> get(const Key& key, bool snapshot = false) override;
+		Future<FDBStandalone<KeyRef>> getKey(const KeySelector& key, bool snapshot = false) override;
+
+		Future<Void> watch(const Key& key) override;
+
+		using Transaction::getRange;
+		Future<FDBStandalone<RangeResultRef>> getRange(const KeySelector& begin, const KeySelector& end,
+													   GetRangeLimits limits = GetRangeLimits(), bool snapshot = false,
+													   bool reverse = false,
+													   FDBStreamingMode streamingMode = FDB_STREAMING_MODE_SERIAL) override;
```

Comment:
This is something we might want to change in the future to a `FutureStream` based API. It's not part of this PR. Also when we do we will add new API instead of changing the existing ones, so it wouldn't break compatibility.

Reply:
Not sure of the history, but I am guessing its to do with `getRange` returning `RangeResultRef` instead of an iterator.

Reply:
Okay, I guess that's fine. Keeping this at `SERIAL` for backwards compatibility's sake is probably fine, then, though we should probably check, say, the document layer for places where we're doing a get range to see if we should be using `ITERATOR` instead.